cv_next/components/layout/navbar.tsx

```diff
@@ -15,98 +18,111 @@ const navLinks = [
   }
 ]
 
-
 export default function Navbar() {
-  const [navbar, setNavbar] = useState(false)
+  const [isNavbarMenuOpen, setIsNavbarMenuOpen] = useState(false)
+  const [isSettignsOpen, setIsSettignsOpen] = useState(false)
+
 
   const handleClick = async () => {
-    setNavbar(false)
+    setIsNavbarMenuOpen(false)
   }
 
   useEffect(() => {
-    if (navbar) {
+    if (isSettignsOpen) {
       document.body.style.overflow = "hidden"
     } else {
       document.body.style.overflow = "auto"
     }
-  }, [navbar])
+  }, [isSettignsOpen])
 
-  return (
-    <header className="select-none">
-      <nav className="mx-auto justify-between px-4 md:flex md:items-center md:px-8 lg:max-w-7xl">
-        <div>
-          <div className="flex items-center justify-between py-3 md:block md:py-5">
-            <Link href="/" onClick={handleClick}>
-              <h1 className="text-2xl font-bold duration-200 lg:hover:scale-[1.10]">
-                CV-NEXT
-              </h1>
-            </Link>
-            <div className="flex gap-1 md:hidden">
-              <button
-                className="text-primary focus:border-primary rounded-md p-2 outline-none focus:border"
-                aria-label="Hamburger Menu"
-                onClick={() => setNavbar(!navbar)}
-              >
-                {navbar ? (
-                  <svg
-                    xmlns="http://www.w3.org/2000/svg"
-                    className="h-6 w-6 "
-                    viewBox="0 0 20 20"
-                    fill="currentColor"
-                  >
-                    <path
-                      fillRule="evenodd"
-                      d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
-                      clipRule="evenodd"
-                    />
-                  </svg>
-                ) : (
-                  <svg
-                    xmlns="http://www.w3.org/2000/svg"
-                    className="h-6 w-6 "
-                    fill="none"
-                    viewBox="0 0 24 24"
-                    stroke="currentColor"
-                    strokeWidth={2}
-                  >
-                    <path
-                      strokeLinecap="round"
-                      strokeLinejoin="round"
-                      d="M4 6h16M4 12h16M4 18h16"
-                    />
-                  </svg>
-                )}
-              </button>
-              <ModeToggle />
+  return <>
```

Comment:
This is a big and messy div hell. Please refactor this component and separate it into smaller ones.

Reply:
Cleaned it up a bit(a lot of unused code, hidden divs, etc)
Also moved the popup trigger to another component and created a navbar dir inside layout to aggregate all related components